### ISSUE TITLE
Relation chain broken

### ISSUE THREAD

**Issue by prodigy7** [robsontenorio/vue-api-query#70]
**Model Setup**
First my models:
Project.js
```
import ModelModule from './ModelModule'
import Product from './Project/Product'

export default class Project extends ModelModule {
  primaryKey() {
    return 'id';
  }

  resource() {
    return 'projects'
  }

  Product() {
    return this.hasMany(Product)
  }
}
```
Project/Product.js
```
import ModelModule from '../ModelModule'
import Storage from './Product/Storage'

export default class Product extends ModelModule {

  primaryKey() {
    return 'id'
  }

  Storage() {
    return this.hasMany(Storage)
  }

  resource() {
    return 'Products';
  }
}
```
Project/Product/Storage.js
```
import ModelModule from '../../ModelModule'

export default class Storage extends ModelModule {
  primaryKey() {
    return 'id';
  }

  resource() {
    return 'Storages'
  }
}
```

**Problem**
```
var product = new Project({
 id: 1
}).Product();
var products = product.find(1);
```
Resulting in `/projects/1/products/1` - so far, so good

```
var products.Storage().get();
```
Resulting in `/products/1/storages` - thats absolutly wrong. Right should be `/projects/1/products/1/storages`
Somehow, the previous part is cut. I know, I can build a workaround with custom. Problem in that situation is, that I pass the model to a general component which should handles different "base models". So I can/want not include api models into the component. My expectation would be, include the models I needed there where I need and pass only the model to the component.
I already tested, it's maybe a error with passing the model to the component by call Storage directly with the response, but I get the same wrong url.

I opened in march already a issue (https://github.com/robsontenorio/vue-api-query/issues/53), but I had missed to answere. Code there had typos and are not reflecting the right code.

I hope my code is right now for showing that issue.

**Comment by prodigy7** [comment]
I build a additional sample:
```
    user = new User({
      id: 10
    }).posts({
      id: 20
    }).comments({
      id: 30
    }).timestamps();
    comment = await user.find(40);
```
I modified model.js from
```
let url = `${this.baseURL()}/${this.resource()}/${this.getPrimaryKey()}/${instance.resource()}`
```
to 
```
let url = `${this.endpoint()}/${instance.resource()}`
```
inside hasMany and expect the path ``http://localhost/users/10/posts/20/comments/30/timestamps/40``, but I got  ``http://localhost/users/10/posts/comments/timestamps/40``.
I now, the id parameters aren't passed. Question is now, how this can made possible.

**Comment by prodigy7** [comment]
I think, I have solved that issue! For made it working, there are some small modifications neccessary. Changes are not breaking changes for methods.

**First**
Edit the model definition file. Sample:

_Before_
```
import BaseModel from './BaseModel'
import Comment from './Comment'

export default class Post extends BaseModel {
  comments () {
    return this.hasMany(Comment)
  }
}
```
_After_ 
```
import BaseModel from './BaseModel'
import Comment from './Comment'

export default class Post extends BaseModel {
  comments (args) {
    return this.hasMany(Comment, args)
  }
}
```

**Second**
Change Model.js method hasMany
```
  hasMany(model) {
    let  instance = new model;
    let url = `${this.baseURL()}/${this.resource()}/${this.getPrimaryKey()}/${instance.resource()}`
    instance._from(url)

    return instance
  }
```
to
```
  hasMany(model, args) {
    let instance;
    if(args === undefined) {
      instance = new model;
    } else {
      instance = new model(args);
    }

    let url = `${this.endpoint()}/${instance.resource()}`

    instance._from(url)

    return instance
  }
```
Now I get the expected result.

**Comment by Peter-Krebs** [comment]
This is related to #16 which was originally resolved with `custom()` and `for()` adjustments.  
The above works when calling relationship methods as well, which is more intuitive.

Needs an update in the readme, too.

**Comment by prodigy7** [comment]
The submitted pull request isn't still complete. Currently I try to fix the path when update/store. Next pull request follows, when fixed

**Comment by prodigy7** [comment]
From my point of view the error is finally solved with pull request #72 when merged
